### DISCUSSION TITLE
Problem with Converting ROMS Variables

### DISCUSSION THREAD

**Discussion by Ibasba** [OpenDrift/opendrift#1357] [General]
Good morning,

In my ROMS file, I have mask_rho while the OpenOil model requires mask_u and mask_v. For the wind, still in my ROMS file, I have wind stress (sustr and svstr), whereas the OpenOil model requires wind (x_wind and y_wind).

Could someone help me resolve this issue?

Thank you

**Comment by zwanghi** [comment]
I am not the expert for the programming issues, but concerning the variables: wind speed and wind stress are two different physical parameters.
Normally you can, depending on sea state, surface roughness,... calculate wind stress from wind speed (that's normally done) the other way is rather unusual. There are plenty of equations on how to do that.
https://rmets.onlinelibrary.wiley.com/doi/abs/10.1002/qj.49708135027
but I would be surprised if ROMS doesn't have wind speed output. or get the meteorological forcing from somewhere else.

Alex

**Comment by AndresSepulveda** [comment]
Well, things have a different name in ROMS/CROCO for some variables. Instead of mask_u and mask_v you have mask_eta and mask_xi

As for u and v velocity, there is no wind speed output since it is wind stress that is used for the calculations. When you input wind speed in a BULK file, it gets converted to wind stress in one of the ways mentioned before (usually Large and Pond, 1981).  

FInd U and V in your BLK input file.

Andrés

**Comment by Ibasba** [comment]
Thank you for your replies. 

As mentioned in my ROMS file, I have the variables mask_rho instead of mask_u and mask_v. Regarding the wind, I have the wind stress (sustr and svstr).
Using the Large and Pond (1981) method, I calculated the wind speed (x_xind and y_wind) which I saved in .mat format.
I used Matlab functions to convert the masks mask_rho to mask_u and mask_v, also saved in .mat format.
My current problem is to integrate these .mat files into the OpenOil model for my entire study period. For each month, I have .mat files containing the wind components (x_wind and y_wind) and .mat files containing the masks (mask_u and mask_v).
How can I configure OpenOil to use these .mat files appropriately?

Thank you in advance for your help.

**Reply by knutfrode** [reply]
You could in principle make a whole new reader for this .mat format, but that would be a lot of work.
Much better would be to save it as netCDF-files, but it must then comply with the ROMS native format.

**Comment by Ibasba** [comment]
Thank you for your response,
I would like to point out that I'm a beginner in Python. I'm more comfortable with Matlab and R, but I hope that by following your instructions, I'll be able to solve the problem.
